packages/components/src/index.native.js

```diff
@@ -6,6 +6,8 @@ export { default as Dashicon } from './dashicon';
 export { default as Dropdown } from './dropdown';
 export { default as Toolbar } from './toolbar';
 export { default as ToolbarButton } from './toolbar-button';
+export { default as ToolbarGroup } from './toolbar-group';
+export { createToolbarSlotFill, ToolbarSlot, ToolbarFill } from './toolbar-slot-fill';
```

Comment:
Is it necessary to expose all those SlotFill related components in the public API?

Reply:
Ideally not. How is the approach to expose things to Gutenberg only?

Reply:
Upon more detailed inspection, I see that only `createToolbarSlotFill` is used in different packages. Isn't it an issue with SlotFill that it doesn't propagate the context properly? There is a PR from @youknowriad where he tries to fix the issue for the Inspector slot: https://github.com/WordPress/gutenberg/pull/16807. I'm sure there is more as I remember that there was PR were another Slot was refactored to bubble virtually and this allowed to fix a similar issue. In fact, it turns out it was targeting the block toolbar, see #16421. I'm sure that Riad will have some good insights on what would be the best way to address it.

Reply:
The related issue which is probably your use case as well: https://github.com/WordPress/gutenberg/issues/9203.

Reply:
The `Slot`/`Fill` module was the trickiest part of this PR for me. Indeed, I agree that those issues should be solved in the module itself. From what I understood (please, correct me if I'm wrong), this is how it currently works:
```jsx
const SlotContext = React.createContext();
const FillContext = React.createContext();

<SlotContext.Provider>
  <Slot name="a" />
  <Slot name="b" bubblesVirtually />
</SlotContext.Provider>

<FillContext.Provider>
  {/* will have access to SlotContext, but not FillContext, because it's rendered in the Slot React tree  */}
  <Fill name="a" />
  {/* will have access to FillContext, but not SlotContext, because it's rendered in this tree and portaled to the slot DOM tree */}
  <Fill name="b" />
</FillContext.Provider>
```

I understand that `bubblesVirtually` was added so `Fill`s could access contexts  in the same tree as custom blocks `edit`, right? Since `BlockControls` uses `bubblesVirtually`, it wouldn't have access to the `ToolbarContext`.

Reply:
The solution I found was passing the context state to the `Fill` tree through the `fillProps` so `Fill` would have access to both `FillContext` and `SlotContext` (`ToolbarContext` in this case).

There may be some way to solve that directly on the slot-fill module, but I'm afraid that it becomes even more frankenstein.

Reply:
Now that I'm thinking about it, putting the ToolbarContext higher in the block component tree would solve this without all the `ToolbarSlot`/`ToolbarFill` components. 😅

Reply:
The principle we try to follow is, if not necessary, let's not add new public APIs because of backward-compatibility reasons. You need to maintain them for years in WordPress world :)

Reply:
I removed `ToolbarSlot`/`ToolbarFill` from `@wordpress/components`. Had to expose `ToolbarContext` though.